packages/asset-swapper/src/swap_quoter.ts

```diff
@@ -410,6 +410,14 @@ export class SwapQuoter {
         return whitelistedApiKeys.includes(apiKey);
     }
 
+    private _isTxOriginBlacklisted(txOrigin: string | undefined): boolean {
+        if (!txOrigin) {
```

Comment:
Are you sure that txOrigin can be undefined?

Reply:
it's explicitly checked for later in the code block